src/query/v2/request_router.hpp

```diff
@@ -719,4 +742,88 @@ class RequestRouter : public RequestRouterInterface {
   io::Notifier notifier_ = {};
   // TODO(kostasrim) Add batch prefetching
 };
+
+class RequestRouterFactory {
```

Comment:
Just a placeholder comment to checkout what we discussed on slack (remove the `variant` and `GetTransportHandle`).